src/compiler/utilities.ts

```diff
@@ -2293,12 +2293,12 @@ namespace ts {
             if (options.jsx === JsxEmit.Preserve) {
                 if (isSourceFileJavaScript(sourceFile)) {
                     if (fileExtensionIs(sourceFile.fileName, ".jsx")) {
-                        extension = ".jsx";
+                        extension = options.jsxExtName || ".jsx";

```

Comment:
@mhegazy can you take a look at this? we as a react-native and ts user need this